src/main/java/Bambot.java

```diff
@@ -0,0 +1,179 @@
+        }
+        if (counter == 0) {
+            System.out.println("List is empty");
+        }
+        System.out.println(DIVIDER);
+    }
+
+    private static void echo() {
+        while (true) {
+            String message = scanner.nextLine();
+            if (handleCommands(message)) return;
+        }
+    }
+
+    // Run different functions base on the first word(Command) in the input
+    private static boolean handleCommands(String message) {
+        String[] commandAndInput = message.split(" ", 2);
+        String command = commandAndInput[0].toLowerCase(); //ensure the command is recognised regardless of capitalisation
+        String input = (commandAndInput.length > 1) ? commandAndInput[1] : "";
+        switch (command) {
+        case "bye":
+            printByeMessage();
+            return true;
+        case "todo":
+            handleToDoCommand(input);
+            break;
+        case "deadline":
+            handleDeadlineCommand(input);
+            break;
+        case "event":
+            try {
+                handleEventCommand(input);
+            } catch (BambotException e) {
+                System.out.println(e.getMessage());
+            }
+            break;
+        case "delete":
+            handleDeleteCommand(input);
+            break;
+        case "list":
+            printList();
+            break;
+        case "mark":
+            handleMarkCommand(input);
+            break;
+        case "unmark":
+            handleUnmarkCommand(input);
+            break;
+        default:
+            System.out.println(message + " is not a valid command");
+            System.out.println(DIVIDER);
+            break;
+        }
+        return false;
+    }
+
+    private static void handleUnmarkCommand(String input) {
+        int unmarkIndex = -1;
+        try {
+            unmarkIndex = Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            System.out.println("Input must be a number (Example: mark 2)");
+        }
+        try {
+            myList.get(unmarkIndex - 1).unmarkTask();
+            printList();
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Index is out of bounds:" + input);
+            System.out.println("Pls use a number in the list");
+        }
+    }
+
+    private static void handleMarkCommand(String input) {
+        int markIndex = -1;
+        try {
+            markIndex = Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            System.out.println("Input must be a number (Example: mark 2)");
+        }
+        try {
+            myList.get(markIndex - 1).markTask();
+            printList();
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Index is out of bounds:" + input);
+            System.out.println("Pls use a number in the list");
+            printList();
+        }
+    }
+
+    private static void handleDeleteCommand(String input) {
```

Comment:
Could consider condensing the try catch block as shown below to make the code more concise:
```suggestion
try {
// logic
} catch (NumberFormatException e) {
// handle NumberFormatException
} catch (IndexOutOfBoundsException e) {
// handle IndexOutOfBoundsException 
}